dio/test/request_integration_test.dart

```diff
@@ -0,0 +1,336 @@
+import 'dart:convert';
+
+import 'package:dio/dio.dart';
+import 'package:test/test.dart';
+
+import 'utils.dart';
+
+void main() {
+  group('requests', () {
+    late Dio dio;
+
+    setUp(() {
+      dio = Dio();
+      dio.options.baseUrl = 'https://httpbun.com/';
+    });
+
+    group('restful APIs', () {
+      const data = {'content': 'I am payload'};
+
+      test('HEAD', () async {
+        final response = await dio.head(
+          '/anything',
+        );
+        expect(response.statusCode, 200);
+        expect(response.isRedirect, isFalse);
+      });
+
+      test('GET', () async {
+        final response = await dio.get(
+          '/get',
+          queryParameters: {'id': '12', 'name': 'wendu'},
+        );
+        expect(response.statusCode, 200);
+        expect(response.isRedirect, isFalse);
+        expect(response.data['method'], 'GET');
+        expect(response.data['args'], {'id': '12', 'name': 'wendu'});
+      });
+
+      // TODO This is not supported on web, should we warn?
+      test('GET with content', () async {
```

Comment:
GET with a body is not supported in web. What should we do here? Exception, warning?

Reply:
We should do nothing. It's under a policy as I remember, and XHR does not ring any bells when the requesting is sending a payload.

Reply:
> It's under a policy as I remember

Not sure what you mean by that.

> XHR does not ring any bells when the requesting is sending a payload

That is correct, the body data is just ignored. The dio API allows this but there is no feedback for the developer why it is not working. My concern is that people open bug reports for this :)

Reply:
> My concern is that people open bug reports for this :)

Adding warning logs like the `ImplyContentTypeInterceptor` might work then.

Reply:
I added a warning